Components/Queues/Handler/PhpAmqpHandler.php

```diff
@@ -0,0 +1,230 @@
+    use IsStopableConsumer;
+
+    /**
+     * @var EndpointInterface
+     */
+    private $endpoint;
+
+    /**
+     * Format used by serializer funtion
+     * @var string
+     */
+    private $format;
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    /**
+     * @var AMQPChannel
+     */
+    private $channel;
+
+    /**
+     * AmqpQueueHandler constructor.
+     *
+     * @param EndpointInterface        $endpoint   The endpoint used to consume
+     * @param int                      $expirationCount  Maximum amount of message to consume before stopping
+     * @param string                   $format     json|xml|php
+     * @param SerializerInterface|null $serializer Serializer to use for deserialization
+     */
+    public function __construct(EndpointInterface $endpoint, int $expirationCount = -1, string $format = 'json', SerializerInterface $serializer = null)
+    {
+        $this->endpoint = $endpoint;
+        $this->setExpirationCount($expirationCount);
+        $this->format = $format;
+        $this->serializer = $serializer ?? SerializerBuilder::create()->build();
+    }
+
+    /**
+     * Consumes the message and dispatch the message to others features
+     * @param string $consumerTag
+     * @param AMQPChannel $channel
+     * @param string $queueName
+     * @throws \Exception
+     */
+    public function consume(string $consumerTag, AMQPChannel $channel, string $queueName): void
+    {
+        $callback = function($message) use ($channel) {
+            $this->isConnected($channel);
+            // Send a message with the string "quit" to cancel the consumer.
+            if ($message->body === 'quit' || $this->shouldStop()) {
+                $channel->basic_cancel($message->delivery_info['consumer_tag']);
+                return false;
+            }
+
+            $this->log('A message was received on {time}');
+            $this->log('Message Body:' . $message->body);
+
+            $queueMessage = $this->deserializeMessage($message);
+            $this->dispatchMessage($queueMessage);
+            $channel->basic_ack($message->delivery_info['delivery_tag']);
+            --$this->expirationCount;
+        };
+
+        try {
+            $channel->basic_qos(null, 1, null);
+            $message = $channel->basic_consume($queue, $consumerTag, false, false, false, false, $callback);
+            $this->isConsuming($channel);
+        } catch (\Exception $exception) {
+            $this->getExceptionHandler()($exception, ['headers' => $message->getHeaders(), 'body' => $message->getBody()]);
+            return;
+        }
+    }
+
+    /**
+     * Verify if the channel is consuming a message.
+     * If there is message to consume it calls the consume callback function
+     * If there is no message to consume it will put the worker in a wait state
+     * @param AMQPChannel $channel
+     */
+    public function isConsuming(AMQPChannel $channel): void
```

Comment:
This method is only used inside the callback. It could be defined as private

Reply:
Done.